### ISSUE TITLE
Feature Request: Rotate selection.

### ISSUE THREAD

**Issue by ProfessorPopoff** [SpaiR/StrongDMM#82]
I'm currently assisting in developing a fresh map for my project, and time and time again i've found that while shuffling buildings around attempting to get them to align, sometimes the best approach is to scrap the building or move it in a place I don't want it in order to fulfil my autistic desires for everything to look somewhat presentable. It'd be wonderful if we were able to use the selection tool and rotate entire areas by 90 degrees.

Complimentary scuffed example:

![image](https://user-images.githubusercontent.com/51188571/100548227-39d95080-3239-11eb-8e77-3b41116e5597.png)
![image](https://user-images.githubusercontent.com/51188571/100548229-3b0a7d80-3239-11eb-86d6-b39f8292bf02.png)
Basically this. (But not quite as low-resolution and scuffed looking!)

**Comment by quardbreak** [comment]
This will be tricky: some objects use different variables for object dir. For example, baystation12 use icon_state as dir on mapping and runtime. If there will be such feature, it must have a configurable local database of variables and patterns to translate values, so that it will be compatible with all repositories.

**Comment by OrangeDecay** [comment]
> This will be tricky: some objects use different variables for object dir. For example, baystation12 use wire cable icon_state as dir on mapping and runtime, like `"0-8"`. If there will be such feature, it must have a configurable local database of variables and patterns to translate values, so that it will be compatible with all repositories.

Even with such flaws it will still do major bulk of the work for you, cleaning it up a little  shouldn't be a big deal, considering how time saving this feature can be.

**Comment by Qumefox** [comment]
I agree. It simply rotating the selection and updating the dir variable accordingly for selected objects/turfs would be extremely time saving. Cleaning up the remaining things needing fixed after rotating would still take less time than completely rebuilding the thing you want to rotate from scratch in a different orientation.
